### PR TITLE
apr: add universal option

### LINKED ISSUE #35811: Subversion fails "symbol(s) not found for architecture i386"
I used "brew install --universal --with-java subversion" to install javahl ,an error appeared.
Any help would be appreciated. Thanks.

wanghaodeMacBook-Pro:~ wanghao$ brew install --universal --with-jdk subversion
==> Downloading http://www.apache.org/dyn/closer.cgi?path=subversion/subversion-
==> Best Mirror http://mirrors.advancedhosters.com/apache/subversion/subversion-
######################################################################## 100.0%
==> Patching
patching file configure
patching file subversion/bindings/swig/perl/native/Makefile.PL.in
==> Downloading https://serf.googlecode.com/svn/src_releases/serf-1.3.7.tar.bz2
######################################################################## 100.0%
==> /usr/local/opt/scons/bin/scons PREFIX=/usr/local/Cellar/subversion/1.8.10_2/
==> /usr/local/opt/scons/bin/scons install
==> ./configure --prefix=/usr/local/Cellar/subversion/1.8.10_2 --with-zlib=/usr 
==> make
      _check_watchdog in named_atomic-test-proc.o
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [subversion/tests/libsvn_subr/named_atomic-proc-test] Error 1
make: *** Waiting for unfinished jobs....

READ THIS: http://git.io/brew-troubleshooting

These open issues may also help:
Subversion fails install on Yosemite (https://github.com/Homebrew/homebrew/issues/35744)
Subversion fails on 10.10 --with-python (https://github.com/Homebrew/homebrew/issues/34119)

This is the debug logs:
https://gist.github.com/anonymous/e123c1dab2ca69e13a84

### PR DESCRIPTION
**Pull request by DomT4**
Closes #35811.

### CODE REVIEW

**Comment by DomT4** [comment]
I don't know if this would be better written less directly and more [like subversion does](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/subversion.rb#L178-L184)? If it would, I'll change it.

File: Library/Formula/apr.rb [hunk 1]
```diff
@@ -23,8 +25,12 @@ def install
     # The internal libtool throws an enormous strop if we don't do...
     ENV.deparallelize
 
+    args = %W[--prefix=#{libexec}]
+
+    args << "--build=i386-apple-darwin" if build.universal?
```
Comment (mikemcquaid) [inline Library/Formula/apr.rb]: Does this still build 64-bit too?

**Comment by jacknagel** [comment]
Did you check to see if simply adding `ENV.universal_binary if build.universal?` is enough? I don't think we want to be explicitly passing `--build` if we can avoid it.

**Comment by DomT4** [comment]
> Did you check to see if simply adding ENV.universal_binary if build.universal?

Yeah, had no affect at all. ` apr ` went right ahead and built for ` x86_64 `.

File: Library/Formula/apr.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/apr.rb]: Yeah, it does an ` uname -m ` to check for the arch at compile time, so it usually automatically pulls it in.

```
checking build system type... x86_64-apple-darwin14.1.0
checking host system type... x86_64-apple-darwin14.1.0
checking target system type... x86_64-apple-darwin14.1.0
Configuring APR library
Platform: x86_64-apple-darwin14.1.0
```

Comment (DomT4) [inline Library/Formula/apr.rb]: With the proposed changes, ` brew install apr --universal ` =

```
checking build system type... i386-apple-darwin
checking host system type... i386-apple-darwin
checking target system type... i386-apple-darwin
Configuring APR library
Platform: i386-apple-darwin
```

**Comment by jacknagel** [comment]
>Yeah, had no affect at all. apr went right ahead and built for x86_64.

How are you confirming this? Don't let the configure output fool you.

I see:

```
$ file /usr/local/Cellar/apr/1.5.1/libexec/lib/libapr-1.dylib
/usr/local/Cellar/apr/1.5.1/libexec/lib/libapr-1.dylib: Mach-O universal binary with 2 architectures
/usr/local/Cellar/apr/1.5.1/libexec/lib/libapr-1.dylib (for architecture i386):	Mach-O dynamically linked shared library i386
/usr/local/Cellar/apr/1.5.1/libexec/lib/libapr-1.dylib (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
```

**Comment by DomT4** [comment]
Oh! Yeah, I was trusting the ` configure ` output to tell me what arch it was building. That's traditionally how I try and spot such things. I had no idea the ` file ` command could reveal a multi-arch binary, that's very cool. Will fix the changes to match the usual formatting, Cheers! :smiley_cat:

**Pull request closed without merging**